File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.kt
fun main() = with(System.`in`.bufferedReader()) {

    // 변수 입력 및 선언
    val n = readLine().toInt()
    val cards = readLine().split(" ").map { it.toInt() }
    val m = readLine().toInt()
    val targets = readLine().split(" ").map { it.toInt() }
    val count = mutableMapOf<Int, Int>()

    // 카운팅
    for (card in cards) {
        count[card] = count.getOrDefault(card, 0) + 1
    }

    // 개수 출력
    val answer = StringBuilder()
    var cnt = 0
    for (target in targets) {
        cnt = count.getOrDefault(target, 0)
        answer.append(cnt).append(" ")
    }

    println(answer)
}

File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.py
import sys

input = sys.stdin.readline

# 변수 및 입력 선언
n = int(input())
arr1 = list(map(int, input().split()))
m = int(input())
arr2 = list(map(int, input().split()))


# O(N) + O(M)
def main():
    # 카드 개수 count O(N)
    count = dict()
    for elem in arr1:
        count[elem] = count.get(elem, 0) + 1

    answer = []
    for elem in arr2:  # O(M)
        count_in_cards = count.get(elem, 0)
        answer.append(count_in_cards)

    print(*answer)


if __name__ == "__main__":
    main()
